### PR TITLE
Refactor build and remove legacy packaging

### PR DESCRIPTION
**Pull request by prateekchaudhry**
### Summary

This PR merges feature branch to remove legacy build code and targets, merges ```INIT_README.md``` and ```README.md``` and rewrites parts of ```README.md```
* (#3529) Unify package build recipes and remove legacy init build targets - Refactors package builds so that all builds use same build scripts. ```dockerfree-agent-image``` renamed as ```release-agent``` as the default agent image build
* (#3535) Remove legacy packaging - Removes legacy builds ```generic-deb```, ```generic-rpm```, ```amazon-linux-ami```. Merges ```INIT_README.md``` content into ```README.md```
* (#3555) Update readme with ECR repo

### Implementation details

Refer (#3529), (#3535) and (#3555) for implementation details

### Testing

Tested Unit tests
Tested Codepipeline builds packages in personal fork

### Description for the changelog

Refactor build and remove legacy packaging

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,28 +2,28 @@
 
 ![Amazon ECS logo](doc/ecs.png "Amazon ECS")
 
-![Build Status](https://github.com/aws/amazon-ecs-agent/workflows/Build/badge.svg?branch=dev)
-
 The Amazon ECS Container Agent is a component of Amazon Elastic Container Service
 ([Amazon ECS](http://aws.amazon.com/ecs/)) and is responsible for managing containers on behalf of Amazon ECS.
 
+This repository comes with ECS-Init, which is a [systemd](http://www.freedesktop.org/wiki/Software/systemd/) based service to support the Amazon ECS Container Agent and keep it running. It is used for systems that utilize `systemd` as init systems and is packaged as deb or rpm. The source for ECS-Init is available in this repository at `./ecs-init` while the packaging is available at `./packaging`.
```
Comment (YashdalfTheGray) [inline README.md:8]: Non-blocking; should we add a comment to the `ecs-init` repo to tell visitors that it has been merged into the ECS agent repo too?

File: README.md [hunk 2]
```diff
@@ -2,28 +2,28 @@
...
 `sudo yum install ecs-init && sudo start ecs`. This is the recommended way to run it in this environment.
 
 ### On Other Linux AMIs
 
+[Amazon ECS docs](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-agent-install.html) provides deb and rpm packages and instructions to install ECS Container Agent on non-Amazon Linux instances.
+
 The Amazon ECS Container Agent may also be run in a Docker container on an EC2 instance with a recent Docker version
 installed. A Docker image is available in our
-[Docker Hub Repository](https://registry.hub.docker.com/u/amazon/amazon-ecs-agent/).
+[Docker Hub Repository](https://hub.docker.com/r/amazon/amazon-ecs-agent).
```
Comment (YashdalfTheGray) [inline README.md:26]: Non-blocking; should we also put the ECR public repo link here? 

https://gallery.ecr.aws/ecs/amazon-ecs-agent

File: scripts/build [hunk 3]
```diff
@@ -29,7 +29,6 @@ with_pause=${4:-false}
 
 PAUSE_CONTAINER_IMAGE="amazon/amazon-ecs-pause"
 PAUSE_CONTAINER_TAG="0.1.0"
-PAUSE_CONTAINER_TARBALL="amazon-ecs-pause.tar"
```
Comment (YashdalfTheGray) [inline scripts/build:32]: Question; I think I missed something, has this line been removed intentionally?

**Review by YashdalfTheGray — COMMENTED** [review]
Thanks for the cleanup and wiring everything properly! This is awesome.

File: scripts/build (hunk 3 above)
Comment (prateekchaudhry) [reply scripts/build:32]: This particular variable is not being used in this script, so I in my opinion it is safe to remove.

File: README.md (hunk 2 above)
Comment (prateekchaudhry) [reply README.md:26]: Thank you for noticing this, should be good to add this

File: README.md (hunk 1 above)
Comment (prateekchaudhry) [reply README.md:8]: We might deprecate repo and as part of that notify that. For now, we do have this [proposal](https://github.com/aws/amazon-ecs-init/issues/482) to notify customers.

Comment (YashdalfTheGray) [reply README.md:8]: Ah, perfect.

**Review by YashdalfTheGray — DISMISSED** [review]

File: README.md (hunk 2 above)
Comment (prateekchaudhry) [reply README.md]: Included in #3555

**Review by YashdalfTheGray — APPROVED** [review]

**Review by mythri-garaga — APPROVED** [review]

**Pull request merged**